src/pubkey.cpp

```diff
@@ -334,6 +336,38 @@ bool CPubKey::Derive(CPubKey& pubkeyChild, ChainCode &ccChild, unsigned int nChi
     return true;
 }
 
+std::optional<EllSqPubKey> CPubKey::EllSqEncode() const {
+    assert(secp256k1_context_verify && "secp256k1_context_verify must be initialized to use CPubKey.");
+
+    secp256k1_pubkey pubkey;
+    if (!secp256k1_ec_pubkey_parse(secp256k1_context_verify, &pubkey, vch, size())) {
+        return {};
+    }
+
+    uint8_t r32[32];
+    GetStrongRandBytes(r32, 32);
```

Comment:
It's a pretty unfortunate I think that the whole random module needs to be included in libbitcoinconsensus just because it include pubkey, and this adds a dependency on randomness.

Alternatives are putting the conversion between pubkey and ellsq encoding in a different modulo than pubkey, or passing in randomness explicitly as argument to `EllSqEncode`.

Also, it's overkill to use strong random bytes here.

Reply:
Done. Passing in random bytes to `EllSqEncode` now.